core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments/activity_items_controller.rb

```diff
@@ -32,11 +27,16 @@ def index
 
     meta = meta(activity_items)
     activity_items = activity_items.map do |activity_item|
-      UffizziCore::Api::Cli::V1::Projects::Deployments::ActivityItemSerializer.new(activity_item).as_json
+      UffizziCore::Api::Cli::V1::Projects::Deployments::ActivityItemSerializer.new(activity_item)
+    end
+
+    if activity_items.any? { |item| item.state == UffizziCore::Event.state.failed }
+      return render json: { errors: { title: [I18n.t('deployment.invalid_state', state: deployment.state)] } },
+                    status: :unprocessable_entity
```

Comment:
Let's return it back. And update this method:
```
    def update_from_compose(compose_file, project, user, deployment, metadata)
      ...
      ActiveRecord::Base.transaction do
        ...
        params = {
          ....
          metadata: deployment_form.metadata,
          state: :active, <<<<---- ADD THIS
        }
        deployment.update!(params)
      end

      deployment
    end
```

I think would be more properly make deployment `active` when we update it

Reply:
This is much better, thanks!